### PR TITLE
Stm32 cube updates

### PR DESCRIPTION
**Pull request by FRASTM**
STM32Cube updates:

stm32cube: update stm32l5 to version V1.4.0
stm32cube: update stm32l4 to version V1.17.0
stm32cube: update stm32g4 to version V1.4.0
stm32cube: update stm32f7 to version V1.16.1
stm32cube: update stm32f4 to version V1.26.0
stm32cube: update stm32g0 to version V1.4.1
stm32cube: update stm32h7 to version V1.9.1  
stm32cube: update stm32mp1 to version V1.4.0  

update stm32cube/common_ll

Signed-off-by: Francois Ramu <francois.ramu@st.com>

### CODE REVIEW

**Comment by FRASTM** [comment]
STM32CubeF7 :  ST Bug tracker ID: 76122 and ST Bug tracker ID: 55274  are now fixed in the v1.16.0
STM32CubeF4 :  ST Bug tracker ID: 76121 is now fixed in the v1.25.2
STM32CubeG0 :  include guards to stm32g0xx_hal_hcd.h in the v1.4.0

**Comment by erwango** [comment]
Thanks for this.
We actually need to skip the MP1 V1.3.0. It triggers compilation issues that will be fixed in V1.4.0 (that should be released in a few if not done yet).

File: stm32cube/stm32g0xx/README [hunk 1]
```diff
@@ -39,6 +39,5 @@ Patch List:
    *Changes from official delivery:
     -dos2unix applied
     -trailing white spaces removed
-    -added include guards to stm32g0xx_hal_hcd.h(STM32CubeG0 GH issue: 13)
```
Comment (str4t0m) [inline stm32cube/stm32g0xx/README:42]: Thanks for the update. As it was not updated in the upstream hal, I think this line should still remain?

Comment (FRASTM) [reply stm32cube/stm32g0xx/README]: right, there was a mistake for stm32g0 and stm32h7

**Comment by FRASTM** [comment]
fixing version change for stm32g0 and stm32h7

**Comment by FRASTM** [comment]
stm32cubeMP1 V1.4.0 is now released --> adding

**Comment by FRASTM** [comment]
adding the stm32mp1 version 1.4.0 --> requires the https://github.com/zephyrproject-rtos/zephyr/pull/33700

**Comment by FRASTM** [comment]
including the stm32mp1 version 1.4.0
compilation ok with `west build -p auto -b stm32mp157c_dk2 samples/hello_world/`

**Comment by str4t0m** [comment]
@FRASTM do you have planned to additionally update the *pcintrl.dtsi files with the latest definitions from STM32 open pin data repo in this or a future pr?

**Comment by erwango** [comment]
> @FRASTM do you have planned to additionally update the *pcintrl.dtsi files with the latest definitions from STM32 open pin data repo in this or a future pr?

@str4t0m this is planned for next week.

**Review by erwango — APPROVED** [review]

**Comment by FRASTM** [comment]
remove remaining trailing space in *.c or .h files

File: stm32cube/stm32mp1xx/CMakeLists.txt [hunk 2]
```diff
@@ -1,4 +1,5 @@
 # Copyright (c) 2020 STMicroelectronics
+# Copyright (c) 2020 STMicroelectronics
```
Comment (ABOSTM) [inline stm32cube/stm32mp1xx/CMakeLists.txt:2]: Useless double Copyright

**Review by ABOSTM — CHANGES_REQUESTED** [review]
Minor comment otherwise LGTM

File: stm32cube/stm32mp1xx/CMakeLists.txt (hunk 2 above)
Comment (FRASTM) [reply stm32cube/stm32mp1xx/CMakeLists.txt]: removed: useless line

**Pull request merged**